apps/evm-profile/src/components/Profile/UploadProfileImage.tsx

```diff
@@ -0,0 +1,49 @@
+import { ProfileMetadata } from '@/types';
+import { IPFS_GATEWAY } from '@/utils/constants';
+import { uploadProfileImage } from '@/utils/profileHelpers';
+import { ChangeEvent, SyntheticEvent, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import Uploader from '../Ui/Uploader';
+
+interface UploadProfileImageProps {
+  value: 'image' | 'profile.banner';
+  innerText: string | undefined;
+  errorMessage?: string;
+}
+
+export default function UploadProfileImage({
+  value,
+  innerText,
+  errorMessage,
+}: UploadProfileImageProps) {
+  const { setValue, trigger } = useFormContext<ProfileMetadata>();
+  const [fileName, setFileName] = useState('');
+  const [isUploadingImage, setIsUploadingImage] = useState(false);
+
+  const uploadImage = async (
+    e: SyntheticEvent | ChangeEvent<HTMLInputElement>,
+  ) => {
+    setIsUploadingImage(true);
+    const result = await uploadProfileImage(e);
+    if (result) {
+      const { hash, fileName } = result;
```

Comment:
Type error here. You need to check if it is a string or object with the `hash` and `fileName` fields. If this is a string then this will blow up